cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -77,6 +78,30 @@ pub mod CometClientComponent {
             comet_client_state.latest_height
         }
 
+        fn update_height_before(
+            self: @ComponentState<TContractState>, client_sequence: u64, target_height: Height
+        ) -> Height {
+            let update_heights = self.read_update_heights(client_sequence);
+
+            let len = update_heights.len();
+
+            assert(len > 0, CometErrors::ZERO_UPDATE_HEIGHTS);
+
+            let mut i = 0;
+
+            while i < len {
+                let height = update_heights.at(i);
```

Comment:
Interesting, do Cairo not have iterators?

Reply:
Something like `filter` or `filter_map` could do that. But a loop is fine for now.